src/game.cpp

```diff
@@ -3365,25 +3365,49 @@ bool game::save_achievements()
         std::replace_copy_if( std::begin( u.name ), std::begin( u.name ) + truncated_name_len,
                               std::ostream_iterator<char>( achievement_file_path ),
         [&]( const char c ) {
-            return !std::isgraph( c, locale );
+            return !std::isgraph( c, locale ) || !is_char_allowed( c );
         }, '_' );
     } else {
-        achievement_file_path << u.name;
+        std::replace_copy_if( std::begin( u.name ), std::begin( u.name ) + truncated_name_len,
```

Comment:
```suggestion
        std::replace_copy_if( std::begin( u.name ), std::end( u.name ),
```
This matches the previous behavior and avoids incorrect truncation of the string in the middle of a utf-8 sequence.

Reply:
Sorry but I am on a time that of a way away from computer, so basically, can't make new commits to these PRs. It may last for quite a time.